File: cpp/Array/L_heightChecker.cpp
#include <bits/stdc++.h>
using namespace std;

int heightChecker(vector<int> &heights)
{
    vector<char> sorted(heights.begin(), heights.end());
    // counting sort
    char freq[101] = {0};
    for (char x : sorted)
        freq[x]++;
    int cnt = 0;
    for (int x = 1; x <= 100; x++)
    {
        int f = freq[x];
        vector<char> f_x(f, x);
        copy(begin(f_x), end(f_x), sorted.begin() + cnt);
        cnt += f;
    }
    // end of count sort

    int n = heights.size(), ans = 0;
    for (int i = 0; i < n; i++)
        ans += (heights[i] != sorted[i]);
    return ans;
}

File: Python/Arrays/L_heightChecker.py
def heightChecker(heights) -> int:
    freq=[0]*101
    for x in heights:
        freq[x]+=1
    cnt=0
    for x in range(1, 101):
        f=freq[x]
        for i in range(cnt, cnt+f):
            heights[i]-=x
        cnt+=f
    return len(heights)-heights.count(0)